conf/i18n/translator.js

```diff
@@ -0,0 +1,156 @@
+const i18next = require('i18next');
+
+/**
+ * This class wraps an instance of the i18next library and provides methods supporting
+ * run-time and compile-time translation. These methods allow for interpolation, pluralization,
+ * and added context.
+ */
+class Translator {
+  /**
+   * Creates a new {@link Translator} that wraps the provided {@link i18next} instance.
+   *
+   * @param {i18next} i18nextInstance The instance to wrap.
+   */
+  constructor (i18nextInstance) {
+    this._i18next = i18nextInstance;
+  }
+
+  /**
+   * Performs a simple translation of the given phrase. If the phrase includes
+   * interpolation, a translated format string, with the relevant placeholders,
+   * is returned.
+   *
+   * @param {string} phrase The phrase to translate.
+   * @returns {string} The translated phrase or format string.
+   */
+  translate (phrase) {
+    const interpPlaceholders = this._getInterpolationPlaceholders(phrase);
+
+    return this._i18next.t(phrase, interpPlaceholders);
+  }
+
+  /**
+   * Provides all the translated singular and plural forms of the given phrase.
+   * The forms will include any of the needed interpolation placeholders.
+   *
+   * @param {string} phrase The phrase to translate.
+   * @param {string} pluralForm The untranslated, plural form of the phrase.
+   * @returns {Object<string|number, string>} A map containing the various forms.
+   *                                          A form is keyed by the
+   *                                          corresponding count or 'plural'.
+   */
+  translatePlural (phrase, pluralForm) {
+    const pluralKeyRegex = new RegExp(`${phrase}_([0-9]+|plural)`);
+
+    const i18nextOptions = this._i18next.options;
+
+    // We first look for the translations in the given locale. If none can be
+    // found there, we iterate through the fallbacks, in order.
+    const localeWithPluralTranslations = this._findLocaleWithTranslationKey(
+      [i18nextOptions.lng, ...i18nextOptions.fallbackLng], pluralKeyRegex);
+
+    if (localeWithPluralTranslations) {
+      const localeTranslations =
+        i18nextOptions.resources[localeWithPluralTranslations].translation;
+
+      // Create a map of count (or 'plural') to the correct translated form.
+      return Object.keys(localeTranslations)
+        .filter(translationKey => pluralKeyRegex.test(translationKey))
+        .reduce(
+          (pluralForms, translationKey) => {
+            const pluralFormIndex = translationKey.split('_')[1];
+            pluralForms[pluralFormIndex] = localeTranslations[translationKey];
+            return pluralForms;
+          },
+          { 1: localeTranslations[phrase] });
+    }
+
+    // If no translations can be found, we return a map containing the provided
+    // singular and plural forms.
+    return {
+      1: phrase,
```

Comment:
oh this is slightly out of date (probably at least part of why runtimeTranslation interpolation wasn't working) I think copying and pasting the translator.js in the most recent jambo localized-pages branch should work though

Reply:
Womp. That makes sense, the danger of duplicating code. Updated!